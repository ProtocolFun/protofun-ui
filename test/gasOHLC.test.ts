import { expect, it } from "vitest";

import { getBuiltGraphSDK } from "../.graphclient";

it("17491129 - gas price for each txn", async () => {
  // arrange
  const sdk = getBuiltGraphSDK();
  // act
  const { txns } = await sdk.TxnsByBlock({
    number: "17491129",
  });
  // assert
  expect(txns.map((x) => x.gasPrice)).toMatchSnapshot();
});

it("17491130 - gas price for each txn", async () => {
  // arrange
  const sdk = getBuiltGraphSDK();
  // act
  const { txns } = await sdk.TxnsByBlock({
    number: "17491130",
  });
  // assert
  expect(txns.map((x) => x.gasPrice)).toMatchSnapshot();
});
