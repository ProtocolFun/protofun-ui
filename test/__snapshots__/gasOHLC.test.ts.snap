// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`17491129 - gas price for each txn 1`] = `
[
  "15270703343",
  "15270703343",
  "15270703343",
  "15270703343",
  "15270703343",
  "15270703343",
  "15293991672",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15320703343",
  "15335703343",
  "15340703343",
  "15350703343",
  "15369468092",
  "15370703343",
  "15370703343",
  "15370703343",
  "15373188032",
  "15373188032",
  "15373188032",
  "15400703343",
  "15445994168",
  "15445994168",
  "15445994170",
  "15470703343",
  "15553015976",
  "15553015976",
  "15616703343",
  "15870703343",
  "15880703343",
  "15880703343",
  "15897515311",
  "16000000000",
  "16000000000",
  "16149629417",
  "16220703343",
  "16220703343",
  "16220703343",
  "16220703343",
  "16220703343",
  "16220703343",
  "16220703343",
  "16220703343",
  "16330666774",
  "16620703343",
  "16620703343",
  "16620703343",
  "16620703343",
  "16620703343",
  "16620703343",
  "16720703343",
  "16720703343",
  "16720703343",
  "16720703343",
  "16720703343",
  "16920703343",
  "16922940770",
  "17108317573",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17220703343",
  "17491358776",
  "17720703343",
  "17720703343",
  "17889641500",
  "18220703343",
  "18220703343",
  "18400000000",
  "19220703343",
  "19220703343",
  "19373188032",
  "20021482301",
  "20504548938",
  "21090000000",
  "22137703343",
  "22137703343",
  "22137703343",
  "22168703343",
  "22491358776",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23851852877",
  "23851852877",
  "25901235251",
  "30553015976",
  "53000000000",
  "15258574920",
  "15258645088",
  "15220703343",
  "17920745412",
  "30000000000",
  "23168991252",
  "53000000000",
  "187531648179",
  "15320703343",
  "15220703343",
]
`;

exports[`17491130 - gas price for each txn 1`] = `
[
  "14684466930",
  "14734466930",
  "14734466930",
  "14734466930",
  "14734466930",
  "14734466930",
  "14734466930",
  "14734466930",
  "14738479495",
  "14738479495",
  "14757755259",
  "14757755259",
  "14984466930",
  "14762135026",
  "14772826930",
  "14772826930",
  "14772826930",
  "14772826930",
  "14783302930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14784466930",
  "14787866930",
  "14800000000",
  "14800000000",
  "14800000000",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14812135026",
  "14834466930",
  "14834466930",
  "14836951619",
  "14836951619",
  "14836951619",
  "14836951619",
  "14836951619",
  "14836951619",
  "14836951619",
  "14836951619",
  "14984466930",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15000000000",
  "15008320337",
  "15008320337",
  "15008320337",
  "15014000000",
  "15014466930",
  "15052800000",
  "15065780512",
  "15210664550",
  "15334466930",
  "15344466930",
  "15684466930",
  "15684466930",
  "15684466930",
  "15684466930",
  "15684466930",
  "15684466930",
  "15684466930",
  "15846771238",
  "15897515311",
  "16000000000",
  "16084466930",
  "16084466930",
  "16084466930",
  "16084466930",
  "16084466930",
  "16084466930",
  "16184466930",
  "16184466930",
  "16184466930",
  "16184466930",
  "16184466930",
  "16184466930",
  "16184466930",
  "16184466930",
  "16330666774",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "16684466930",
  "17109466930",
  "17184466930",
  "17184466930",
  "17184466930",
  "17616706213",
  "17616706213",
  "17684466930",
  "17684466930",
  "18181450954",
  "18400000000",
  "18684466930",
  "18836951619",
  "18836951619",
  "19684466930",
  "19684466930",
  "19684466930",
  "21753466930",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "23000000000",
  "64684466930",
  "14722280551",
  "14784466930",
  "23000000000",
  "23000000000",
  "30000000000",
  "24627390498",
  "15684466930",
  "14757755259",
  "18324844011",
  "14684466930",
]
`;
